subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -33,56 +34,82 @@ class DeviceWorker(
     fun run() = scope.launch {
 
         var state: State = try {
-            checkDeviceAlive()
-            device.state()
-        } catch (t: Exception) {
-            device.warn("Error while checking device initial state", t)
-            // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
-            // Can't use this device any more. ReservationClient will get a new one.
-            return@launch
+            when (val status = device.deviceStatus()) {
+                is Device.DeviceStatus.Alive -> device.state()
+                is Device.DeviceStatus.Freeze -> {
+                    // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
+                    // Can't use this device any more. TODO Will ReservationClient get a new one?
+                    device.warn("Device wasn't booted. Failed on initial run", status.reason)
+                    return@launch
+                }
+            }
+        } catch (t: Throwable) {
+            throw RuntimeException("Unexpected error when initialize a $device", t)
         }
 
+        // what will happen when worker dies
         for (intention in intentionsRouter.observeIntentions(state)) {
             try {
-                checkDeviceAlive()
-
                 device.log("Receive intention: $intention")
-
-                device.log("Preparing state: ${intention.state}")
-                state = prepareDeviceState(currentState = state, intentionState = intention.state).get()
-                device.log("State prepared")
-
-                val result = executeAction(action = intention.action)
-                device.log("Worker test run completed for intention")
-                messagesChannel.send(
-                    DeviceWorkerMessage.Result(
-                        intentionResult = IntentionResult(
-                            intention = intention,
-                            actionResult = InstrumentationTestRunActionResult(
-                                testCaseRun = result
-                            )
-                        )
-                    )
-                )
-
+                when (val status = device.deviceStatus()) {
+                    is Device.DeviceStatus.Alive -> {
+                        device.log("Preparing state: ${intention.state}")
+                        val (preparingError, newState) = prepareDeviceState(
```

Comment:
Let's discuss this. We have two ways
single `catch` that reacts on each exception equally or we explicitly handle each exception
I prefer to handle everything explicitly and fail on Unexpected errors

Reply:
> I prefer to handle everything explicitly and fail on Unexpected errors

Agree, it seems safer